Mathlib/Data/Nat/Lattice.lean

```diff
@@ -66,6 +66,12 @@ theorem iInf_of_empty {ι : Sort*} [IsEmpty ι] (f : ι → ℕ) : iInf f = 0 :=
   rw [iInf_of_empty', sInf_empty]
 #align nat.infi_of_empty Nat.iInf_of_empty
 
+@[simp]
+lemma iInf_const_zero {ι : Sort*} : ⨅ i : ι, 0 = 0 := by
+  cases isEmpty_or_nonempty ι
+  · exact iInf_empty _
+  · exact ciInf_const
```

Comment:
```suggestion
    (isEmpty_or_nonempty ι).elim (fun h => by simp) (fun h => sInf_eq_zero.2 <| by simp)
```
if you'd like. The proof with `ciInf_const` doesn't seem to work unless it's below the instance defined later.